File: client-clojure/src/sirqul_io_t_platform/specs/application_mini_response.clj
(ns sirqul-io-t-platform.specs.application-mini-response
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.name-string-value-response :refer :all]
            [sirqul-io-t-platform.specs.asset-short-response :refer :all]
            [sirqul-io-t-platform.specs.asset-short-response :refer :all]
            )
  (:import (java.io File)))


(def application-mini-response-data
  {
   (ds/opt :valid) boolean?
   (ds/opt :message) string?
   (ds/opt :version) float?
   (ds/opt :serializeNulls) boolean?
   (ds/opt :startTimeLog) int?
   (ds/opt :errorCode) string?
   (ds/opt :request) (s/coll-of name-string-value-response-spec)
   (ds/opt :applicationId) int?
   (ds/opt :appKey) string?
   (ds/opt :title) string?
   (ds/opt :currentBuildVersion) string?
   (ds/opt :currentApiVersion) string?
   (ds/opt :appIcon) asset-short-response-spec
   (ds/opt :appLogo) asset-short-response-spec
   (ds/opt :active) boolean?
   (ds/opt :returning) string?
   })

(def application-mini-response-spec
  (ds/spec
    {:name ::application-mini-response
     :spec application-mini-response-data}))

File: client-scalaz/src/main/scala/org/openapitools/client/api/ApplicationMiniResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import ApplicationMiniResponse._

case class ApplicationMiniResponse (
  valid: Option[Boolean],
message: Option[String],
version: Option[Double],
serializeNulls: Option[Boolean],
startTimeLog: Option[Long],
errorCode: Option[String],
request: Option[List[NameStringValueResponse]],
applicationId: Option[Long],
appKey: Option[String],
title: Option[String],
currentBuildVersion: Option[String],
currentApiVersion: Option[String],
appIcon: Option[AssetShortResponse],
appLogo: Option[AssetShortResponse],
active: Option[Boolean],
returning: Option[String])

object ApplicationMiniResponse {
  import DateTimeCodecs._

  implicit val ApplicationMiniResponseCodecJson: CodecJson[ApplicationMiniResponse] = CodecJson.derive[ApplicationMiniResponse]
  implicit val ApplicationMiniResponseDecoder: EntityDecoder[ApplicationMiniResponse] = jsonOf[ApplicationMiniResponse]
  implicit val ApplicationMiniResponseEncoder: EntityEncoder[ApplicationMiniResponse] = jsonEncoderOf[ApplicationMiniResponse]
}
